private/buf/cmd/buf/command/curl/curl.go

```diff
@@ -497,6 +545,125 @@ func (f *flags) validate(isSecure bool) error {
 	return nil
 }
 
+func (f *flags) determineCredentials(
+	ctx context.Context,
+	container interface {
+		app.Container
+		appverbose.Container
+	},
+	host string,
+) (string, error) {
+	if f.UserCreds != "" {
+		// this flag overrides any netrc-related flags
+		var username, password string
+		pos := strings.IndexByte(f.UserCreds, ':')
```

Comment:
nit: can use `strings.SplitN`

Reply:
Yeah, I guess I'm in a habit of `strings.Index` to avoid the slice allocation. But that's definitely premature optimization for this bit of code. Done.